File: api/src/endpoint/runner/create_runner.rs
use serde::{Deserialize, Serialize};

use api_derive::ApiRequest;

use crate::{endpoint::Endpoint, resource::RunnerResource};

pub struct CreateRunner;

impl Endpoint for CreateRunner {
    const PATH: &'static str = "/ci/runner/{owner}";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateRunnerRequest;
    type Response = CreateRunnerResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateRunnerRequest {
    pub name: String,
    pub owner_type: String,
}

pub type CreateRunnerResponse = RunnerResource;

File: gitdot-api-ts/src/endpoint/runner/create-runner.ts
import { z } from "zod";

import { RunnerResource } from "../../resource";

export const CreateRunnerRequest = z.object({
  name: z.string(),
  owner_type: z.string(),
});
export type CreateRunnerRequest = z.infer<typeof CreateRunnerRequest>;

export const CreateRunner = {
  path: "/ci/runner/{owner}",
  method: "POST",
  request: CreateRunnerRequest,
  response: RunnerResource,
} as const;
export type CreateRunner = typeof CreateRunner;
